lib/stanford/storage_object_validator.rb

```diff
@@ -0,0 +1,63 @@
+require 'moab'
+
+module Stanford
+  # druids are Stanford specific entities
+  class StorageObjectValidator < Moab::StorageObjectValidator
+    # error codes
+    EXPECTED_DATA_SUB_DIRS = ["content", "metadata"].freeze
+    IMPLICIT_DIRS = ['.', '..'].freeze # unlike Find.find, Dir.entries returns these
+    DATA_DIR_NAME = "data".freeze
+    EXPECTED_VERSION_SUB_DIRS = [DATA_DIR_NAME, "manifests"].freeze
+    MANIFEST_INVENTORY_PATH = 'manifests/manifestInventory.xml'.freeze
+    SIGNATURE_CATALOG_PATH = 'manifests/signatureCatalog.xml'.freeze
+
+    INCORRECT_DIR = 0
+    MISSING_DIR = 1
+    EXTRA_CHILD_DETECTED = 2
+    VERSION_DIR_BAD_FORMAT = 3
+    NO_SIGNATURE_CATALOG = 4
+    NO_MANIFEST_INVENTORY = 5
+    NO_XML_FILES = 6
+    VERSIONS_NOT_IN_ORDER = 7
+    FILES_IN_VERSION_DIR = 8
+    DRUID_MISMATCH = 9
```

Comment:
if you wanted to get fancy, perhaps you could do

```
# note that next line assumes keys are in ascending numerical order
DRUID_MISMATCH = ERROR_CODE_TO_MESSAGES.keys.last + 1
```

not sure if that's better than simply hard-coding a 9.